Src/Lecoati.LeBlender.Extension/Models/LeBlenderPropertyModel.cs

```diff
@@ -28,36 +30,23 @@ public class LeBlenderPropertyModel
 
         public T GetValue<T>()
         {
-
+			var helper = new Helper();
             //var targetContentType = Helper.GetTargetContentType();
-            var targetDataType = Helper.GetTargetDataTypeDefinition(Guid.Parse(DataTypeGuid));
-
-            var propertyType = new PublishedPropertyType(Helper.GetTargetContentType(),
-                new PropertyType(new DataTypeDefinition(targetDataType.PropertyEditorAlias)
-                {
-                    Id = targetDataType.Id
-                }));
-
-            // Try Umbraco's PropertyValueConverters
-            var converters = PropertyValueConvertersResolver.Current.Converters.ToArray();
-            foreach (var converter in converters.Where(x => x.IsConverter(propertyType)))
-            {
-                // Convert the type using a found value converter
-                var value2 = converter.ConvertDataToSource(propertyType, Value, false);
-
-                // If the value is of type T, just return it
-                if (value2 is T)
-                    return (T)value2;
-
-                // If ConvertDataToSource failed try ConvertSourceToObject.
-                var value3 = converter.ConvertSourceToObject(propertyType, value2, false);
-
-                // If the value is of type T, just return it
-                if (value3 is T)
-                    return (T)value3;
-
-                // Value is not final value type, so try a regular type conversion aswell
-                var convertAttempt = value2.TryConvertTo<T>();
+            var targetDataType = helper.GetTargetDataTypeDefinition(Guid.Parse(DataTypeGuid));
+
+			// This propertyType is a mock, where only the TargetDatatype.EditorAlias is used in x.IsConverter(propertyType)
+			// This constructor is the one with minimal validity checks. See comment in PublishedPropertyType.cs.
+			// But this may change in future, so be prepared to look into the Umbraco source code, if this code fails.
+			var propertyType = new PublishedPropertyType( "pt-" + targetDataType.Id, targetDataType.Id, true, ContentVariation.Nothing, new PropertyValueConverterCollection( new IPropertyValueConverter[] { } ), new PublishedModelFactoryMock(), Current.PublishedContentTypeFactory );
+
+			var converters = Current.Factory.GetInstance<PropertyValueConverterCollection>().ToArray();
+			foreach (var converter in converters.Where( x => x.IsConverter( propertyType ) ))
+			{
+				// Since the ConvertDataToSource and ConvertSourceToObject methods don't exist anymore,
+				// We skip the code and try to convert the Value property directly.
+
+				// Value is not final value type, so try a regular type conversion aswell
+				var convertAttempt = Value.TryConvertTo<T>();
```

Comment:
This might work with simple types, but not e.g. ´.GetValue<IPublishedContent>` (content picker, media picker (single), etc.)

Maybe have a look at the core property value converters https://github.com/umbraco/Umbraco-CMS/tree/853087a75044b814df458457dc9a1f778cc89749/src/Umbraco.Core/PropertyEditors

Not sure if there is an recommended approach to get value converter supported in Umbraco core, but there are something like this in core:
https://github.com/umbraco/Umbraco-CMS/blob/853087a75044b814df458457dc9a1f778cc89749/src/Umbraco.Core/PropertyEditors/PropertyValueConverterCollection.cs#L30-L32

@soreng do you have any thoughs on this?